### ISSUE TITLE
Doesn't work: the server responded with a status of 429

### ISSUE THREAD

**Issue by andykras** [lencx/nofwl#72]
Fresh install via `brew`

[Error] Failed to load resource: the server responded with a status of 429 (Bad Request) (completions, line 0)

![image](https://user-images.githubusercontent.com/5300156/236679062-315780d1-5ae2-4dbc-ad4d-444067e47fbc.png)

[NoFWL.log](https://github.com/lencx/nofwl/files/11415018/NoFWL.log)

nofwl.conf.json
```json
{
  "lang": "en",
  "stay_on_top": false,
  "theme": "dark",
  "titlebar": false,
  "main_init": true,
  "main_ua": "",
  "main_close": false,
  "main_width": 800.0,
  "main_height": 600.0,
  "main_url": "https://ai.com",
  "tray": true,
  "tray_ua": "",
  "tray_width": 360.0,
  "tray_height": 540.0,
  "tray_url": "https://ai.com"
}
```

gpt.conf.json
```json
{
  "api_key": "*****",
  "api_model": "gpt-3.5-turbo",
  "side_bar": true,
  "side_menu": true,
  "active_chat": "*****",
  "api_context_level": 100
}
```

**Comment by andykras** [comment]
I found it in responses of `completions `. It seems that I exceed openai quota:
```json
{
    "error": {
        "message": "You exceeded your current quota, please check your plan and billing details.",
        "type": "insufficient_quota",
        "param": null,
        "code": null
    }
}
```
It would be nice to have some more clear message:
![image](https://user-images.githubusercontent.com/5300156/236679383-609454a9-7886-44e1-9b34-62d429b00aa7.png)

**Comment by Piscabo** [comment]
As of Today:
![image](https://github.com/lencx/nofwl/assets/123719842/303ca518-2f0e-4e48-91c5-2e04a223b22c)

{
    "error": {
        "message": "You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY), or as the password field (with blank username) if you're accessing the API from your browser and are prompted for a username and password. You can obtain an API key from https://platform.openai.com/account/api-keys.",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
API Key has been provided:
![image](https://github.com/lencx/nofwl/assets/123719842/3f204acb-491a-4898-af8e-0bbae59d788b)

**Comment by andykras** [comment]
Right, but my point was that the message "Request failed with status code 429" is not clear. It would be helpful to put a details in the application logs. Or even tell user exactly what happened (in my case it was quota exceed). Thanks!

**Comment by Piscabo** [comment]
Wasn't downplaying your comment, was indicating my problem. Yes, would be nice to know where to look to fix this. Mine still fails  429, I ran Lighthouse, and here is the report, someone please help? There seems to be no activity on this repository!?
[tauri.localhost-20230516T121305.pdf](https://github.com/lencx/nofwl/files/11486818/tauri.localhost-20230516T121305.pdf)

**Comment by Piscabo** [comment]
> Right, but my point was that the message "Request failed with status code 429" is not clear. It would be helpful to put a details in the application logs. Or even tell user exactly what happened (in my case it was quota exceed). Thanks!

How did you sort this out, traced it to the same problem:
![image](https://github.com/lencx/nofwl/assets/123719842/8c224db9-63d9-418f-a9a9-4b1d20d7d4a0)

The 'readme' should make clear then that NoFWL is a PAID service?!

**Comment by jraxxo** [comment]
> The 'readme' should make clear then that NoFWL is a PAID service?!


what? no it's not.
NoFWL isn't even a service, it's a desktop app calling the openAI api (which IS a service) in the background so it can answer your requests.
the openAI api, in turn, does offer some free usage but is quite limited. you should consider switching to a pay-as-you-go plan if you need to be able to make more requests than the free tier allows. 
it's still quite cheap.

**Comment by Piscabo** [comment]
> > The 'readme' should make clear then that NoFWL is a PAID service?!
> 
> what? no it's not. NoFWL isn't even a service, it's a desktop app calling the openAI api (which IS a service) in the background so it can answer your requests. the openAI api, in turn, does offer some free usage but is quite limited. you should consider switching to a pay-as-you-go plan if you need to be able to make more requests than the free tier allows. it's still quite cheap.

Is it 'making use' of a service that uses tokens that you have to pay for, otherwise it won't work, yes it is, I rest my case.

**Comment by jraxxo** [comment]
rest your case all you want, doesn't make your statements any more correct. 
nofwl is still just a frontend for a service which is being provided by a third party. 
and yes, you CAN use it for free. 
openAI gives every user $18 or so worth of tokens initially. 
what, do you expect @lencx to use their own api key and pay for all requests made through nofwl? 
just be grateful that there's people like them who make stuff like this for free in their spare time. 
god, you're so entitled, it's disgusting.

**Comment by lencx** [comment]
This project has currently only released a simplest version, and there are still a lot of features under development. In the future, it will also support web access, just like the ChatGPT desktop application.

**Comment by Piscabo** [comment]
> rest your case all you want, doesn't make your statements any more correct. nofwl is still just a frontend for a service which is being provided by a third party. and yes, you CAN use it for free. openAI gives every user $18 or so worth of tokens initially. what, do you expect @lencx to use their own api key and pay for all requests made through nofwl? just be grateful that there's people like them who make stuff like this for free in their spare time. god, you're so entitled, it's disgusting.

Listen here, NO reason to take out your Bitchy attitude on me. Call me disgusting, who the hell are you? They give $5 in South Africa, I haven't asked a few questions, and the tokens were depleted. I simply was pointing, if it stated as a token system or error 429 at least pointed it out, I would have not questioned it. Keep your Nasty comments to yourself, please! You don't know me or my situation! Talking about self-entitlement look in the mirror! I appreciate everything the open-source community puts forward, and help where I can. Sorry @lencx for this.

**Comment by jraxxo** [comment]
k

**Comment by peter9811** [comment]
Same problem here. I don't know how to fix it

**Comment by jrapp2** [comment]
@peter9811 you need to switch to a pay-as-you-go plan on [the OpenAI page](https://openai.com/pricing), as your free credits have very likely run out.

**Comment by jrapp2** [comment]
@peter9811 you need to switch to a pay-as-you-go plan on [the OpenAI page](https://openai.com/pricing), as your free credits have very likely run out.
